test/cqlpy/test_name.py

```diff
@@ -37,63 +45,196 @@ def passes_or_raises(expected_exception, match=None):
+# or rejected with an InvalidRequest exception (Scylla/Cassandra 3).
+# The error message must contain the name of the view, so that we can
+# verify that the test is actually testing the name length limit, and not
+# some other error.
+def test_mv_name_length_gt_than_scylla_limit(cql, test_keyspace):
+    name = padded_name(MAX_NAME_LENGTH + 1)
+    with new_test_table(cql, test_keyspace, SCHEMA) as table:
+        with passes_or_raises(InvalidRequest, match=name):
+            with new_materialized_view(
+                cql,
+                table,
+                "*",
+                "p, x",
+                "p is not null and x is not null",
+                name=name,
+            ):
+                pass
+
+
+# Verifies that a secondary index name of exactly 207 characters is accepted.
+def test_index_name_length_eq_scylla_limit(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, SCHEMA) as table:
+        with new_secondary_index(cql, table, "x", padded_name(MAX_NAME_LENGTH)):
             pass
 
-# TODO: add tests for the allowed characters in a table name
-# TODO: add tests like we have above for table names also for keyspace
-# names, index names, function names, column names, etc.
+
+# Verifies that a secondary index name longer than 207 characters is accepted (Cassandra 4/5)
+# or rejected with an InvalidRequest exception (Scylla/Cassandra 3).
+# The error message must contain the name of the index, so that we can
+# verify that the test is actually testing the name length limit, and not
+# some other error.
+def test_index_name_length_gt_than_scylla_limit(cql, test_keyspace):
+    name = padded_name(MAX_NAME_LENGTH + 1)
+    with new_test_table(cql, test_keyspace, SCHEMA) as table:
+        with passes_or_raises(InvalidRequest, match=name):
+            with new_secondary_index(cql, table, "x", name):
+                pass
+
+
+# Verifies that a column name of exactly 207 characters is accepted.
+def test_column_name_length_eq_scylla_limit(cql, test_keyspace):
+    with new_test_table(
+        cql,
+        test_keyspace,
+        "p int primary key, " + padded_name(MAX_NAME_LENGTH) + " int",
+    ):
+        pass
+
+
+# Verifies that a column name longer than 207 characters is accepted (Cassandra 4/5)
+# or rejected with an InvalidRequest exception (Scylla/Cassandra 3).
+# The error message must contain the name of the column, so that we can
+# verify that the test is actually testing the name length limit, and not
+# some other error.
+def test_column_name_length_gt_than_scylla_limit(cql, test_keyspace):
+    name = padded_name(MAX_NAME_LENGTH + 1)
+    with passes_or_raises(InvalidRequest, match=name):
+        with new_test_table(cql, test_keyspace, "p int primary key, " + name + " int"):
+            pass
+
+
+# Verifies that function names are not limited by MAX_NAME_LENGTH (can be much longer).
+def test_function_name_length(cql, test_keyspace):
```

Comment:
This is a Scylla-only test (Cassandra doesn't have lua), so please add scylla_only.

Please try to run this entire test file on Cassandra (with test/cqlpy/run-cassandra) to see it passes.